src/producer.rs

```diff
@@ -216,26 +179,57 @@ fn zip_producer(tmp_dir: &Path, zip_files: &[&String], queue: &WorkQueue, ignore
 }
 
 pub fn producer(tmp_dir: &Path, paths: &[String], queue: &WorkQueue, ignore_orphan_gcno: bool, is_llvm: bool) -> Option<Vec<u8>> {
-    let mut zip_files = Vec::new();
     let mut directories = Vec::new();
 
+    let mut gcno_archive: Option<ZipArchive<File>> = None;
+    let mut gcda_archives: Vec<(&String, Archive)> = Vec::new();
+    let mut info_archives: Vec<(&String, ZipArchive<File>)> = Vec::new();
```

Comment:
If we make `gcno_archive` and `info_archives` `Archive` objects too, we might be able to do some cleanups.